registry/proxy/proxyauth.go

```diff
@@ -19,7 +19,7 @@ type credentials struct {
 }
 
 func (c credentials) Basic(u *url.URL) (string, string) {
-	up := c.creds[u.String()]
+	up := c.creds[tokenURL]
```

Comment:
It was not my intent to imply we are/should be limiting to token auth here. I was merely stating we should be limiting the credentials to the token auth server when token auth is used. When basic auth is used the credentials should be used for the `remoteURL`. This particular part of the code is not making any decision about auth type and is only using credentials that are intended to be used for the provided `*url.URL`.